adapta/storage/query_enabled/_models.py

```diff
@@ -0,0 +1,146 @@
+"""
+ Query Enabled Store Connection interface.
+"""
+
+#  Copyright (c) 2023. ECCO Sneaks & Data
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+
+import re
+from abc import ABC, abstractmethod
+from enum import Enum
+from pydoc import locate
+from typing import TypeVar, Generic, Type, Iterator, Union, final, Optional
+
+import pandas
+
+from adapta.storage.models.base import DataPath
+from adapta.storage.models.filter_expression import Expression
+
+TCredential = TypeVar("TCredential")  # pylint: disable=C0103
+TSettings = TypeVar("TSettings")  # pylint: disable=C0103
+
+CONNECTION_STRING_REGEX = r"^qes:\/\/class=(.*?);plaintext_credentials=(.*?);settings=(.*?)$"
+
+
+@final
+class BundledQes(Enum):
+    """
+    QES Implementations aliases that are bundled with Adapta.
+    """
+
+    DELTA = "adapta.storage.query_enabled.DeltaQes"
+    ASTRA = "adapta.storage.query_enabled.AstraQes"
+
+
+BUNDLED_STORES = {store.name: store.value for store in BundledQes}
+
+
+class QueryEnabledStore(Generic[TCredential, TSettings], ABC):
+    """
+    QES base class.
+    """
+
+    def __init__(self, credentials: TCredential, settings: TSettings):
```

Comment:
This class can validate that the `credentials` object is supported by the store. For instance it can throw an error if you try to run a query through AstraQes with Delta credentials.

Reply:
Creds are validated when `.from_json` is called on a credential class via `post_init` - will add test to prove the claim